### PR TITLE
Wallet connect init

### LINKED ISSUE #2683: Initialize wallet connect setup
- [ ] create a feature flag
- [ ] add Wallet Connect service behind a feature flag
- [ ] install the dependencies
- [ ] add initial logic to connect to a dApp
  - > It's expected at this point to have a lot of hard-coded parts and a lot of concerns not handled at all

### PR DESCRIPTION
**Pull request by greg-nagy**
### What
This PR includes a very early [WalletConnect](https://walletconnect.com/) integration logic, with related dependencies and feature flag.
Its current goal is to perform basic bootstrap operations and initiate a (type v2) connection, via hardcoded parameters.

Closes #2683

### Test precursors
To be able to perform tests, it is necessary to **apply** the following patch file
[test.patch.txt](https://github.com/tallycash/extension/files/10123819/test.patch.txt)
using:
`git apply test.patch.txt`

This patch adds console printouts that serve as test indicators,  bypassing linter limitation that blocked the build

In addition, before test execution, it is also necessary to set a valid, hardcoded, WalletConnect connection URI value in the [wcUri constant](https://github.com/tallycash/extension/blob/579257ad12f892240435ad8b687b2f1a2f73870b/background/services/wallet-connect/index.ts#L80)
This URI can be retrieved by initiating a connection against any WalletConnect v2 client, as seen in this video:
https://user-images.githubusercontent.com/15876717/204826666-320ba172-9ce0-4dea-8bb2-f82f29f27b9a.mov

### Testing
Please the following demo dApp client, on Ethereum [sample dApp](https://react-dapp-v2-with-ethers.vercel.app/) to acquire a WalletConnect URI
Paste into the [wcUri constant](https://github.com/tallycash/extension/blob/579257ad12f892240435ad8b687b2f1a2f73870b/background/services/wallet-connect/index.ts#L80)

On extension load or restart, in extensions background console log:
- [ ] Observe WalletConnect initialization as `Core Initilization Success` and `SignClient Initilization Success` messages
- [ ] Observe connection request as a `[WalletConnect Demo] - pairing request sent` message
- [ ] Observe connection success as a `[WalletConnect Demo] - connection acknowledged` message
- [ ] Note any additional WalletConnect protocol system and keep-alive messages such as:
image

Latest build: [extension-builds-2679](https://github.com/tallycash/extension/suites/9616582216/artifacts/459073835) (as of Thu, 01 Dec 2022 18:42:07 GMT).

### CODE REVIEW

File: background/features.ts [hunk 1]
```diff
@@ -31,6 +31,7 @@ export const RuntimeFlag = {
   SUPPORT_ACHIEVEMENTS_BANNER:
     process.env.SUPPORT_ACHIEVEMENTS_BANNER === "true",
   SUPPORT_NFT_TAB: process.env.SUPPORT_NFT_TAB === "true",
+  SUPPORT_WALLET_CONNECT: process.env.SUPPORT_NFT_TAB === "true",
```
Comment (mhluongo) [inline background/features.ts:34]: Nope

**Review by mhluongo — COMMENTED** [review]
Draft thoughts for a draft PR

File: background/features.ts (hunk 1 above)
Comment (greg-nagy) [reply background/features.ts]: [bcca153](https://github.com/tallycash/extension/pull/2679/commits/bcca153f302460a50e35cd059b1d33f87f28451a)

File: background/main.ts [hunk 2]
```diff
@@ -288,6 +290,18 @@ export default class Main extends BaseService<never> {
...
+    const walletConnectService = isEnabled(FeatureFlags.SUPPORT_WALLET_CONNECT)
+      ? WalletConnectService.create(
+          providerBridgeService,
+          internalEthereumProviderService,
+          preferenceService
+        )
+      : (Promise.resolve({
+          startService: () => Promise.resolve(),
+          stopService: () => Promise.resolve(),
+          emitter: new Emittery(),
```
[the remark this answers is not in the document]
Comment (greg-nagy) [reply background/main.ts]: [5e787c9](https://github.com/tallycash/extension/pull/2679/commits/5e787c98c2b5bf2de932081e66b412a53134b5d1)

**Comment by greg-nagy** [comment]
done in #2707

**Pull request closed without merging**